AEM/intel-extension-for-openxla.md

```diff
@@ -0,0 +1,249 @@
+
+<p align="center">Figure 2: Intel® Extension for OpenXLA* Architecture</p>
+
+## Enabling Intel® Extension for OpenXLA\* 
+
+In this section, we show how to enable JAX applications on Intel GPUs with Intel® Extension for OpenXLA\*.
+
+### Installation
+
+1.	Preinstallation Requirements
+
+   * Install Intel GPU driver
+
+      Intel® Data Center GPU Max Series [627.7 driver](https://dgpu-docs.intel.com/releases/stable_627_7_20230526.html)
+
+   * Install Intel® oneAPI Base Toolkit
+
+      Install oneAPI Base Toolkit Packages [2023.1](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit.html)
+
+   * Setup environment variables
+   ```bash
+   source /opt/intel/oneapi/setvars.sh  # Modify based on your oneAPI installation directory
+   ```
+   * Install bazel
+   ```bash
+   wget https://github.com/bazelbuild/bazel/releases/download/5.3.0/bazel-5.3.0-installer-linux-x86_64.sh
+   bash bazel-5.3.0-installer-linux-x86_64.sh --user
+   bazel –-version   # Verify bazel version is 5.3.0
+   ```
+2.	Create a Python\* Virtual Environment (using Miniconda)
+
+   * Install [Miniconda\*](https://docs.conda.io/en/latest/miniconda.html)
+
+   * Create and activate virtual running environment
+   ```bash
+   conda create -n jax python=3.9
+   conda activate jax
+   pip install --upgrade pip
+   ```
+3. Build and install JAX and Intel® Extension for OpenXLA
+   ```bash
+   git clone https://github.com/intel/intel-extension-for-openxla.git
+   cd intel-extension-for-openxla
+   pip install jax==0.4.7 jaxlib==0.4.7
+   ./configure
+   bazel build //xla: pjrt_plugin_xpu.so
+   ```
+4. Set library paths
+   ```bash
+   export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"/python-path/lib/python3.9/site-packages/jaxlib/"
+   export PJRT_NAMES_AND_LIBRARY_PATHS="xpu:/xla-path/bazel-bin/xla/pjrt_plugin_xpu.so"
+   ```
+### Check That PJRT Intel GPU Plugin Is Loaded
+
+Use a Python call to `jax.local_devices()` to check all available XPU devices:
+```bash
+python -c "import jax;print(jax.local_devices())"
+```
+Sample output: on Intel® Data Center GPU Max 1550 GPU (with 2 stacks):
+```
+[xpu(id=0), xpu(id=1)]
+```
+In this case of a server with single Intel® Data Center GPU Max 1550 GPU installed, there are two XPU devices, representing 2 stacks loaded into the current process. 
+
+>**Note**: If there is no XPU devices detected, the output would be as below:
+```
+No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
+[CpuDevice(id=0)]
+```
+In this case, make sure both the environment variables for `oneAPI` and the library paths for `jaxlib` as well as `PJRT plugin` have been set as described in above “Installation” section.
+
+## Running Stable Diffusion in JAX on Intel GPUs
+
+### Install Required Packages
+```bash
+pip install flax==0.6.6 transformers==4.27.4 diffusers==0.16.1 datasets==2.12.0
+sudo apt install numactl
```

Comment:
looks like that we need Ubuntu.
Should we mention OS requirement in Preinstallation Requirement session?

Reply:
Done